### PR TITLE
Authentication test

### PR DESCRIPTION
**Pull request by munterkalmsteiner**

### CODE REVIEW

File: bundles/specmate-auth-test/src/com/specmate/auth/test/InMemorySessionServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		context = FrameworkUtil.getBundle(InMemorySessionServiceTest.class).getBundleContext();
+		sessionService = getSessionService();
+	}
+	
+	@Test
+	public void testIsAuthorized() throws SpecmateException {
+		String token = sessionService.create(AccessRights.ALL, "test");
+		assertTrue(sessionService.isAuthorized(token, "localhost/services/rest/test/resource1"));
+		assertTrue(sessionService.isAuthorized(token, "localhost/services/rest/test/resource1/resource2"));
+		assertTrue(sessionService.isAuthorized(token, "localhost/services/rest/test/"));
```
Comment (junkerm) [inline bundles/specmate-auth-test/src/com/specmate/auth/test/InMemorySessionServiceTest.java]: why is there a difference in the result of the former and the latter?

File: bundles/specmate-auth-test/src/com/specmate/auth/test/InMemorySessionServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	
+	@BeforeClass
+	public static void init() throws Exception {
+		context = FrameworkUtil.getBundle(InMemorySessionServiceTest.class).getBundleContext();
+		sessionService = getSessionService();
+	}
+	
+	@Test
+	public void testIsAuthorized() throws SpecmateException {
+		String token = sessionService.create(AccessRights.ALL, "test");
```
[the remark this answers is not in the document]
Comment (munterkalmsteiner) [reply bundles/specmate-auth-test/src/com/specmate/auth/test/InMemorySessionServiceTest.java]: Fixed

File: bundles/specmate-auth-test/src/com/specmate/auth/test/InMemorySessionServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		assertTrue(sessionService.isAuthorized(token, "localhost/services/rest/test/resource1/resource2"));
+		assertTrue(sessionService.isAuthorized(token, "localhost/services/rest/test/"));
+		assertFalse(sessionService.isAuthorized(token, "localhost/services/rest/test"));
+		assertFalse(sessionService.isAuthorized(token, "localhost/services/rest/"));
+		assertFalse(sessionService.isAuthorized(token, "localhost/services/rest"));
+	}
+	
+	@Test
+	public void testRegexInjection() throws SpecmateException {
+		String token = sessionService.create(AccessRights.ALL, "test");
```
[the remark this answers is not in the document]
Comment (munterkalmsteiner) [reply bundles/specmate-auth-test/src/com/specmate/auth/test/InMemorySessionServiceTest.java]: Fixed

File: bundles/specmate-auth-test/src/com/specmate/auth/test/InMemorySessionServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+		
+		token = sessionService.create(AccessRights.ALL, ".+");
+		assertFalse(sessionService.isAuthorized(token, "localhost/services/rest/pro/resource1"));
+	}
+	
+	@Test
+	public void testDeleteSession() throws SpecmateException {
+		boolean thrown = false;
+		String token = sessionService.create(AccessRights.ALL, "test");
+		assertTrue(sessionService.isAuthorized(token, "localhost/services/rest/test/resource1"));
```
[the remark this answers is not in the document]
Comment (munterkalmsteiner) [reply bundles/specmate-auth-test/src/com/specmate/auth/test/InMemorySessionServiceTest.java]: Fixed

**Pull request merged**